contracts/Kasu.sol

```diff
@@ -92,17 +93,41 @@ contract Kasu is KasuStorage {
 
     // [Feature 3] Borrower's dashboard
     // borrower can see all the NFTs they borrowed
-    function viewRentedListings(uint256 listingId) public {
-
+    function viewRentedListings(address borrowerAddress) public {
+        // Iterating with a guard is expesnive...
     }
 
     // [Feature 3] Borrower's dashboard
     // After borrower return NFT, collateral is sent from smart contract to borrower's address
     function returnNFT(uint256 listingId) public {
-
+        Listing memory listing = _getListingById(listingId);
+        require(
+            _assertOwnership(
+                listing.tokenAddress,
+                listing.lenderAddress,
+                listing.tokenId
+            ) == true,
+            "Token not returned yet"
+        );
```

Comment:
I understand that you're trying to check if the NFT has been transferred to the original owner. But ideally we want this to be an atomic transaction:
1. call IERC721.safeTransferFrom() to transfer the NFT (assuming the contract has been approved)
2. transfer collateral to the borrower
3. transfer interest to the lender
4. delete the listing from storage.

Therefore you can make sure there are no loose ends (e.g. what if the borrower returns the NFT, but didn't call returnNFT() and was still charged interest?)